src/main/java/com/google/sps/servlets/ImageServlet.java

```diff
@@ -59,19 +57,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     byte[] blobBytes = getBlobBytes(blobKey);
     List<EntityAnnotation> annotations = getImageLabels(blobBytes);
-    writer.println("Annotations size: " + annotations.size());
-    writer.println("Blob Key: " + blobKey);
-    writer.println("Number of bytes: " + blobBytes.length);
-   
-    // Create an array of labels for printing the appropriate JSON 
-    List<String> imageLabels = new ArrayList<>();;
-    for (int i = 0; i < imageLabels.size() && i < MAX_NUM_KEYWORDS; i++) {
-      imageLabels.add(annotations.get(i).getDescription());
-    }
+    Keywords.addKeywords(annotations);
+
+    Set<String> keywords = Keywords.getKeywords();
     Gson gson = new Gson();
-    String json = gson.toJson(imageLabels);
-    writer.println(json);
-    response.sendRedirect("/main.html");
+    String json = gson.toJson(keywords);
+    response.sendRedirect(String.format("/results?k=%s", Keywords.KEY));
```

Comment:
I'm not sure if I'm understanding correctly but keywords.KEY is a constant value, and what should be passed to the URL is the datastore key of the object, so it can be retrieved later. This is just passing a constant URL for every set of keywords.

Reply:
By object I mean the set of keywords extracted from every single input.

Reply:
How do you suggest creating keys for each of these sets of keywords? Because the key has to come from somewhere. As it is right now, I'm storing each of the keywords and their salience as entities via the same key (as defined by KEY), so I think the only way to change this is to use user authentication and use the user ID to differentiate between each user's set of keywords.

Reply:
Every element that is added to datastore has a unique key assigned to them.